haystack/document_stores/memory.py

```diff
@@ -281,47 +281,57 @@ def get_document_by_id(
         else:
             return None
 
-    def get_documents_by_id(self, ids: List[str], index: Optional[str] = None) -> List[Document]:  # type: ignore
+    def get_documents_by_id(
+        self,
+        ids: List[str],
+        index: Optional[str] = None,
+        batch_size: Optional[int] = None,
+        headers: Optional[Dict[str, str]] = None,
+    ) -> List[Document]:
         """
         Fetch documents by specifying a list of text id strings.
         """
+        if headers:
+            raise NotImplementedError("InMemoryDocumentStore does not support headers.")
+        if batch_size:
+            logger.warning(
+                "InMemoryDocumentStore does not support batching in `get_documents_by_id` method. This parameter is ignored."
+            )
         index = index or self.index
         documents = [self.indexes[index][id] for id in ids]
         return documents
 
-    def get_scores_torch(self, query_emb: np.ndarray, document_to_search: List[Document]) -> List[float]:
+    def get_scores_torch(self, query_emb: np.ndarray, documents_to_search: List[Document]) -> List[float]:
```

Comment:
The renaming of parameter to `documents_to_search ` in the `get_scores`, `get_scores_torch`, and `get_scores_numpy` methods is not a breaking change because all three methods aren't user facing. Otherwise we would have needed to add the `breaking change` tag to the PR but it's all good.